README.md

```diff
@@ -137,30 +137,34 @@ let who = game.whose_turn(); // white/black
 // show a visual representation of chess, through stdout
 game.show_board();
 
-// to select a character
-let character = game.select(Pawn(1, G)); // warns if not legal
+// to select a character //if this engine doesn't provide GUI selection replace select with move
+let character = game.select(Pawn(1, G)); // warns if not legal 
 
-// list out possible moves for selected character.
+// List out possible moves for selected character.
 let moves = character.possible_moves();
 
-// move character to new position,
-// in case of valid move, it spits out effect
-// otherwise throws a error.
+// Move the character to a new position,
+// In case of a valid move, it spits out the effect
+// otherwise throws an error.
 // also makes the timer stop.
 let res = character.move_to(2, G); // ok/err
 
-// can't select character, if you dont' switch side.
-game.select(Pawan(2, G)); // warns since turn over.
+// can't select a character, if you don't switch side.
+game.select(Pawn(2, G)); // warns since turn over.
 
 // changes the side to get select working again.
 // also starts a counter.
 game.change_side();
 
 let who = game.whose_turn(); // white/black
 
-game.accept_draw();
+game.request_draw();
+
+game.resign();
 
 game.is_game_over();
 
 game.check_winner();
+
+game.can_pawn_promote();
```

Comment:
it would be done roughly as:

let pawn = game.select(Pawn(7,B)); // could be any
let res = pawn.move_to(8, B);
// `res` would suggest mandatory `promote`, and enable `promote` to work,
pawn.promote(); // disabled initially